pallets/parachain-staking/src/lib.rs

```diff
@@ -2048,6 +2054,14 @@ pub mod pallet {
 			let actual_weight = T::WeightInfo::delegator_bond_more(
 				<DelegationScheduledRequests<T>>::get(&candidate).len() as u32,
 			);
+
+			// If we have called delegator_bond_more for this deletagor before, the weight
+			// returned will be an over-estimate since the read was already performed on the first
+			// call and subsequent calls do not increase PoV size further.
+			if already_called_for_delegator {
+				actual_weight.set_proof_size(0);
```

Comment:
Despite its name, `set_proof_size` does not mutate in place

```suggestion
				actual_weight = actual_weight.set_proof_size(0);
```

Reply:
https://paritytech.github.io/substrate/master/src/sp_weights/weight_v2.rs.html#46 seems okay, this is a `mut self`

Reply:
Nope, `&mut self` mutates in place, `mut self` mutates a copy.